lib/commands/sync.js

```diff
@@ -192,12 +202,36 @@ module.exports = function *(argv) {
     let queue = blockingQueue()
 
     console.log(`Synchronizing ${totalRecords} records in ${table}...                                                                        `)
-    let sourceCursor = yield sr.db(sourceDB).table(table).orderBy({index: sr.asc('id')})
-      .map(function (row) { return {id: row('id'), hash: sr.uuid(row.toJSON())} })
-      .run({cursor: true})
-    let targetCursor = yield tr.db(targetDB).table(table).orderBy({index: tr.asc('id')})
-      .map(function (row) { return {id: row('id'), hash: tr.uuid(row.toJSON())} })
-      .run({cursor: true})
+    let sourceCursor
+    let targetCursor
+    if (companyIds) {
+      console.log(`Synchronizing by Companies: ${companyIds}`)
+      sourceCursor = yield sr.db(sourceDB).table(table)
+        .orderBy({index: "id"})
+        .filter(el =>
+          sr.expr(companyIds)
+          .contains(el("Company"))
+        )
+        .map(function (row) { return {id: row('id'), hash: sr.uuid(row.toJSON())} })
+        .run({cursor: true})
+      targetCursor = yield tr.db(targetDB).table(table)
+        .orderBy({index: "id"})
+        .filter(el =>
+          tr.expr(companyIds)
+          .contains(el("Company"))
+        )
+        .map(function (row) { return {id: row('id'), hash: tr.uuid(row.toJSON())} })
+        .run({cursor: true})
+    } else {
+      console.log(`Synchronizing everything`)
+      sourceCursor = yield sr.db(sourceDB).table(table)
+        .orderBy({index: sr.asc('id')})
+        .map(function (row) { return {id: row('id'), hash: sr.uuid(row.toJSON())} })
+        .run({cursor: true})
+      targetCursor = yield tr.db(targetDB).table(table).orderBy({index: tr.asc('id')})
+        .map(function (row) { return {id: row('id'), hash: tr.uuid(row.toJSON())} })
+        .run({cursor: true})
+    }
```

Comment:
This needs to be DRY'ed up. The only difference from the original code is adding the filter if company IDs are passed in

Reply:
DRY'ed